src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -270,11 +410,374 @@ class DropStatementSegment(BaseSegment):
+                    Sequence(
+                        "TERMINATED", "BY", Ref("QuotedLiteralSegment"), optional=True
+                    ),
+                    Sequence(
+                        Ref.keyword("OPTIONALLY", optional=True),
+                        "ENCLOSED",
+                        "BY",
+                        Ref("QuotedLiteralSegment"),
+                        optional=True,
+                    ),
+                    Sequence(
+                        "ESCAPED", "BY", Ref("QuotedLiteralSegment"), optional=True
+                    ),
+                    optional=True,
+                ),
+                Sequence(
+                    "LINES",
+                    Sequence(
+                        "STARTING", "BY", Ref("QuotedLiteralSegment"), optional=True
+                    ),
+                    Sequence(
+                        "TERMINATED", "BY", Ref("QuotedLiteralSegment"), optional=True
+                    ),
+                    optional=True,
+                ),
+            ),
+        ),
+    )
+
+
+# I wanted to do an override and just insert the new Refs but I need
+# to have the IntoClause be before FromClauseSegment, but it doesn't work to add the `before` clause
+# looking for suggestions on how to do this differently, if possible
+@mysql_dialect.segment(replace=True)
+class UnorderedSelectStatementSegment(BaseSegment):
+    """A `SELECT` statement without any ORDER clauses or later.
+
+    This is designed for use in the context of set operations,
+    for other use cases, we should use the main
+    SelectStatementSegment.
+    """
+
+    type = "select_statement"
+    # match grammar. This one makes sense in the context of knowing that it's
+    # definitely a statement, we just don't know what type yet.
+    match_grammar = StartsWith(
+        # In mysql, the select clause may include an INTO statement
+        # to assign values from columns/functions to corresponding
+        # local or session variables
+        Ref("SelectClauseSegment"),
+        terminator=OneOf(
+            Ref("SetOperatorSegment"),
+            Ref("WithNoSchemaBindingClauseSegment"),
+            Ref("OrderByClauseSegment"),
+            Ref("LimitClauseSegment"),
+            Ref("NamedWindowSegment"),
+        ),
+        enforce_whitespace_preceeding_terminator=True,
+    )
+
+    parse_grammar = Sequence(
+        Ref("SelectClauseSegment"),
+        # Dedent for the indent in the select clause.
+        # It's here so that it can come AFTER any whitespace.
+        Dedent,
+        Ref("IntoClauseSegment", optional=True),
+        Ref("FromClauseSegment", optional=True),
+        Ref("WhereClauseSegment", optional=True),
+        Ref("GroupByClauseSegment", optional=True),
+        Ref("HavingClauseSegment", optional=True),
+        Ref("ForClauseSegment", optional=True),
+    )
+
+
+# I am unclear why I have to override this segement, but if I don't then new segments won't parse
```

Comment:
The issue is that when a segment is overridden, the parser looks for all Ref() segments within the present dialect. To fix this, you need to explicitly refer to the segments in the source dialect from which you are overriding. See https://github.com/sqlfluff/sqlfluff/blob/master/src/sqlfluff/dialects/dialect_postgres.py#L195 for an example.

Reply:
I am not clear on your suggestion.  I would like to completely remove the overrides for `SelectClauseElementSegment`, `SelectClauseSegment`, and `SelectStatementSegment` since I am not altering them at all.  Could you provide a specific example of what you are asking me to change?

Reply:
IIUC, Niall is saying (for example), that at line 678, you should replace this code:
```
            Ref("SelectClauseElementSegment"),
```

with something like this:
```
            ansi_dialect.get_segment("SelectClauseElementSegment"),
```

I haven't encountered this situation before, so I'm not sure this is exactly right. Can you give it a try and see if it works? If it does, you should be able to remove lines 624-650 (the copy pasted `SelectClauseElementSegment` from the ANSI dialect).

And once we get this working, a similar technique will allow removing the other segments mentioned above.